template/.github/workflows/cd.yml

```diff
@@ -8,107 +8,92 @@ on:
 jobs:
 {% if crate_type == "bin" %}
   publish:
-    name: Publishing for {{ "${{ matrix.job.os }}" }}
+    name: Publishing for {{ "${{ matrix.job.os-name }}" }}-{{ "${{ matrix.job.architecture }}" }}
     runs-on: {{ "${{ matrix.job.os }}" }}
     permissions:
       contents: write
     strategy:
+      fail-fast: false
       matrix:
         rust: [stable]
         job:
           - os: macos-latest
             os-name: macos
             target: x86_64-apple-darwin
             architecture: x86_64
-            binary-postfix: ""
-            use-cross: false
           - os: macos-latest
             os-name: macos
             target: aarch64-apple-darwin
             architecture: arm64
-            binary-postfix: ""
-            use-cross: false
           - os: ubuntu-latest
             os-name: linux
             target: x86_64-unknown-linux-gnu
             architecture: x86_64
-            binary-postfix: ""
-            use-cross: false
           - os: windows-latest
             os-name: windows
             target: x86_64-pc-windows-msvc
             architecture: x86_64
-            binary-postfix: ".exe"
-            use-cross: false
           - os: ubuntu-latest
             os-name: linux
             target: aarch64-unknown-linux-gnu
             architecture: arm64
-            binary-postfix: ""
-            use-cross: true
           - os: ubuntu-latest
             os-name: linux
             target: i686-unknown-linux-gnu
             architecture: i686
-            binary-postfix: ""
-            use-cross: true
 
     steps:
       - name: Checkout repository
         uses: actions/checkout@v4
       - name: Install Rust toolchain
-        uses: actions-rs/toolchain@v1
+        uses: dtolnay/rust-toolchain@v1
         with:
           toolchain: {{ "${{ matrix.rust }}" }}
           target: {{ "${{ matrix.job.target }}" }}
-          profile: minimal
-          override: true
-      - uses: Swatinem/rust-cache@v2
-      - name: Cargo build
-        uses: actions-rs/cargo@v1
+      - name: Install cross
+        uses: taiki-e/install-action@v2
         with:
-          command: build
-          use-cross: {{ "${{ matrix.job.use-cross }}" }}
-          toolchain: {{ "${{ matrix.rust }}" }}
-          args: --release --target {{ "${{ matrix.job.target }}" }}
+          tool: cross
+      - name: Build release binary
+        run: cross build --locked --release --target {{ "${{ matrix.job.target }}" }}
```

Comment:
should we only use and install cross if `use-cross: true`? this will make `cd` more efficient